gui/default/syncthing/core/syncthingController.js

```diff
@@ -1343,16 +1358,30 @@ angular.module('syncthing.core')
         $scope.directoryList = [];
 
         $scope.$watch('currentFolder.path', function (newvalue) {
-            if (newvalue && newvalue.trim().charAt(0) === '~') {
-                $scope.currentFolder.path = $scope.system.tilde + newvalue.trim().substring(1);
+            if (!(newvalue)) {
+                return;
             }
+            $scope.currentFolder.path = expandTilde(newvalue);
             $http.get(urlbase + '/system/browse', {
                 params: { current: newvalue }
             }).success(function (data) {
                 $scope.directoryList = data;
             }).error($scope.emitHTTPError);
         });
 
+        $scope.$watch('currentFolder.label', function (newvalue) {
+            if (!$scope.config.options || !$scope.config.options.defaultFolderPath || $scope.currentFolder.path) {
+                return;
+            }
+            var path = $scope.config.options.defaultFolderPath;
+            if (newvalue) {
+                path = pathJoin(path, newvalue);
+            } else {
+                path = pathJoin(path, $scope.currentFolder.id);
+            }
+            $('#folderPath').prop("value", path);
```

Comment:
As it's implemented now, I think it introduces a bug where if none of the fields get touched, currentFolder.path ends up being empty.

Also, it doesn't handle the case where the ID is changed by the user, yet the label is left blank. It will stick with the first ID it ever sees.

So there is four ways to solve this:

1. Use $dirty/$pristine checks on currentFolder.path to decide if you want to keep updating it as user tweaks the id or label. (here)[https://stackoverflow.com/questions/18641618/how-can-i-denote-which-input-fields-have-changed-in-angularjs]

2. Keep updating path every time user changes the label/ID. Hopefully they settle on what it's supposed to look like before meddling with the path (annoying)

3. Do what you did, but somehow patch up the empty path case.

4. Something like this:
```
    $scope.$watchGroup(['currentFolder.label', 'currentFolder.id'], function (newvalues, oldvalues) {
              if (!$scope.config.options || !$scope.config.options.defaultFolderPath || $scope.currentFolder.path) {
                  return;
              }
              oldvalue = oldvalues[0] || oldvalues[1]; // equivalnet to .label || .id
              newvalue = newvalues[0] || newvalues[1];

              // Label being changed, as path at old label matches current path, hence update with new ID/label.
              // In all other cases it's the path b
              if (pathJoin($scope.config.options.defaultFolderPath, oldvalue) != $scope.currentFolder.path) {
                   $scope.currentFolder.path = pathJoin($scope.config.options.defaultFolderPath, newvalue);
              }
    }
```
This way you avoid the jQuery magic.